modyn/tests/storage/internal/file_wrapper/binary_file_wrapper_test.cpp

```diff
@@ -24,30 +25,70 @@ class BinaryFileWrapperTest : public ::testing::Test {
       : config_{StorageTestUtils::get_dummy_file_wrapper_config()},
         filesystem_wrapper_{std::make_shared<MockFilesystemWrapper>()} {
     file_name_ = tmp_dir_ + "/test.bin";
+    file_name_endian_ = tmp_dir_ + "/test_endian.bin";
   }
 
   void SetUp() override {
     std::filesystem::create_directory(tmp_dir_);
 
     std::ofstream file(file_name_, std::ios::binary);
-    const std::vector<std::pair<uint32_t, uint16_t>> data = {{42, 12}, {43, 13}, {44, 14}, {45, 15}};
+    const std::vector<std::pair<uint16_t, uint16_t>> data = {{42, 12}, {43, 13}, {44, 14}, {45, 15}};
     for (const auto& [payload, label] : data) {
       payload_to_file(file, payload, label);
     }
     file.close();
+
+    // tmp test file with 2 byte labels
+    std::ofstream file_endian(file_name_endian_, std::ios::binary);
+    const std::vector<std::pair<uint16_t, uint16_t>> data_endian = {
+        {(1u << 8u) + 2u, 0}, {(3u << 8u) + 4u, 1}, {(5u << 8u) + 6u, 2}, {(7u << 8u) + 8u, 3}};
+    for (const auto& [payload, label] : data_endian) {
+      // note: on macOS & linux, the architecture's endian is little
+      payload_to_file(file_endian, payload, label);
```

Comment:
This does not depend on the OS, but on the processor. For example, you can run Linux on IBM POWER CPUs that are big endian. I am not sure whether the test currently assumes that it's always run on a LE machine, but in the best case, we would not have such implicit assumptions

Reply:
That's a valid point, the test file endian depends on the system running the tests. Running the test file creation on big-endian would yield different results. I will make the file creation endian independent by writing bytes instead of `uint16_t`s